File: Die Newton Raphson Methode/Newton_Raphson_Methode.java
public class Approximation {
    public static void main(String[] args) {
        double n = 9.0;
        double guess = n / 2.0;
        double tolerance = 0.0001;

        while (Math.abs(guess * guess - n) > tolerance) {
            guess = (guess + n / guess) / 2.0;

        System.out.println("Approximierte Quadratwurzel von " + n + " = " + guess);
    }
}

File: Die Newton Raphson Methode/Newton_Raphson_Methode.py
def square_root(n):
    guess = n/2  
    tolerance = 0.0001  

    while abs(guess**2 - n) > tolerance:
        guess = (guess + n/guess) / 2 

    return guess

print(square_root(9))
print(square_root(25))
print(square_root(2))
